parquet-tools/src/main/java/org/apache/parquet/tools/command/MergeCommand.java

```diff
@@ -65,14 +75,28 @@ public void execute(CommandLine options) throws Exception {
 
     // Merge schema and extraMeta
     FileMetaData mergedMeta = mergedMetadata(inputFiles);
+    PrintWriter out = new PrintWriter(Main.out, true);
 
     // Merge data
     ParquetFileWriter writer = new ParquetFileWriter(conf,
             mergedMeta.getSchema(), outputFile, ParquetFileWriter.Mode.CREATE);
     writer.start();
+    boolean tooSmallFilesMerged = false;
     for (Path input: inputFiles) {
+      if (input.getFileSystem(conf).getFileStatus(input).getLen() < TOO_SMALL_FILE_THRESHOLD) {
+        out.format("Warning: file %s is too small, length: %d\n",
+          input,
+          input.getFileSystem(conf).getFileStatus(input).getLen());
+        tooSmallFilesMerged = true;
+      }
+
       writer.appendFile(conf, input);
     }
+
+    if (tooSmallFilesMerged) {
+      out.println("Warning: you merged too small files. " +
+        "Despite the size of the merged file is bigger, it STILL contains small row groups, thus you don't have the advantage of big row groups!");
```

Comment:
I would use stronger wording after the last comma, like "which usually leads to bad query performance." (Additionally, I'm not a native speaker either, but I believe you either need a "that" after "Despite" or use some other word, for example "Although".)